Formula/s/spotifyd.rb

```diff
@@ -35,24 +26,63 @@ class Spotifyd < Formula
   depends_on "pkgconf" => :build
   depends_on "rust" => :build
   depends_on "dbus"
-  depends_on "portaudio"
+
+  on_macos do
+    depends_on "portaudio"
+  end
+
+  on_linux do
+    depends_on "alsa-lib"
+    depends_on "pulseaudio"
+  end
+
+  def backend
+    on_macos do
+      return "portaudio"
+    end
+    on_linux do
+      return "pulseaudio"
+    end
+  end
 
   def install
     ENV["COREAUDIO_SDK_PATH"] = MacOS.sdk_path_if_needed if OS.mac?
 
-    system "cargo", "install", "--no-default-features",
-                               "--features", "dbus_keyring,portaudio_backend",
-                               *std_cargo_args
+    # The same features as upstream's "default" feature set binaries
+    # https://github.com/Spotifyd/spotifyd/blob/master/.github/workflows/cd.yml
+    features = %w[dbus_mpris rodio_backend]
+    features += if OS.mac?
+      %w[portaudio_backend]
+    else
+      %w[alsa_backend pulseaudio_backend]
+    end
+
+    system "cargo", "install", "--no-default-features", "--features", features.join(","), *std_cargo_args
+
+    inreplace "contrib/spotifyd.conf", /^#backend = "alsa"/, "backend = \"#{backend}\""
+    etc.install "contrib/spotifyd.conf"
+  end
+
+  def caveats
+    "The service's configuration can be modified at #{etc}/spotifyd.conf"
   end
 
   service do
-    run [opt_bin/"spotifyd", "--no-daemon", "--backend", "portaudio"]
+    run [opt_bin/"spotifyd", "--no-daemon", "--config-path", etc/"spotifyd.conf"]
```

Comment:
Hardcoding the config path will probably lead to confusion. Especially when people update several formulae it's easy to miss the caveats.